marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/provenance/dhProvLibTest.sjs

```diff
@@ -0,0 +1,107 @@
+'use strict';
+declareUpdate();
+const provLib = require('/data-hub/5/provenance/dh-provenance.xqy');
+const test = require("/test/test-helper.xqy");
+const jobService = require("../lib/jobService.sjs");
+const config = require("/com.marklogic.hub/config.sjs");
+
+function verifyProvenanceRecord() {
+  const id = "1ca9230c-7d83-43b8-bbe3-ef98c43e2989";
+  const recordOpts = {
+    "startTime": "2021-06-10T04:00:00",
+    "user": xdmp.getCurrentUser()
+  }
+
+  const record = provLib.provenanceRecord(id, recordOpts);
+  return [
+    test.assertEqual("job:1ca9230c-7d83-43b8-bbe3-ef98c43e2989", xs.string(fn.head(record.xpath("/*:activity/@*:id")))),
+    test.assertEqual("1ca9230c-7d83-43b8-bbe3-ef98c43e2989", xs.string(fn.head(record.xpath("/*:activity/*:label")))),
+    test.assertEqual("dh:Job", xs.string(fn.head(record.xpath("/*:activity/*:type")))),
+    test.assertEqual("2021-06-10T04:00:00", xs.string(fn.head(record.xpath("/*:activity/*:startTime")))),
+    test.assertEqual("user:test-data-hub-developer", xs.string(fn.head(record.xpath("/*:agent/@*:id")))),
+    test.assertEqual("test-data-hub-developer", xs.string(fn.head(record.xpath("/*:agent/*:label")))),
+    test.assertEqual("dh:User", xs.string(fn.head(record.xpath("/prov:agent/prov:type")))),
+    test.assertEqual("job:1ca9230c-7d83-43b8-bbe3-ef98c43e2989", xs.string(fn.head(record.xpath("/*:wasAssociatedWith/*:activity/@*:ref")))),
+    test.assertEqual("user:test-data-hub-developer", xs.string(fn.head(record.xpath("/*:wasAssociatedWith/*:agent/@*:ref"))))
+  ];
+}
+
+function verifyProvenanceRecordCreatedAfterStartJob() {
+  const jobId = "verifyProvenanceRecordCreatedAfterStartJob-job-id-1";
+  const flowName = "verifyProvenanceRecordCreatedAfterStartJob-flow-1";
+  const jobStatus = "finished";
+  xdmp.invokeFunction(function () {
+    declareUpdate();
+    jobService.startJob(jobId, flowName);
+  });
+  const uris = queryProvenanceRecord(jobId);
+  let record = fn.head(fn.doc(uris.finalRecordUri));
+
+  return [
+    test.assertNotEqual(null, cts.doc(uris.stagingRecordUri)),
+    test.assertNotEqual(null, cts.doc(uris.finalRecordUri)),
+    test.assertEqual("job:verifyProvenanceRecordCreatedAfterStartJob-job-id-1", xs.string(fn.head(record.xpath("/*:document/*:activity/@*:id")))),
```

Comment:
For this test and any test that involves XML data, I'd strongly consider using XQuery for the tests. These assertions are pretty difficult to read because of the bloat from "xs.string / fn.head / record.xpath". An equivalent assertion in XQuery would be much nicer to read:

```
let $activity := $record/*:document/*:activity
return (
  test:assert-equal("dh:Job", $activity/*:label/text())
```

And I also wouldn't use a wildcard in the expressions above. A wildcard should only be used when we don't know what the namespace might be. We know exactly what's expected here, so we should use the specific namespace prefix for each element. 

Finally - you could write the test in SJS, and then just do all the assertions in an XQuery block of code. I think that'd still be a lot less code and a lot easier to read.